anvio/synteny.py

```diff
@@ -0,0 +1,303 @@
+        genes_and_functions_list = [] # List of lists [gene-caller-id, accessions, contig-name]
+        counter = 0
+        for gci in genes_in_contigs: # gci = gene-caller-id
+            list_of_gene_attributes = []
+
+            if gci in gci_to_accession_dict:
+                accession = gci_to_accession_dict[gci]
+                accession = accession.replace(" ","")
+                contig_name = genes_in_contigs[gci]['contig']
+                list_of_gene_attributes.extend((gci, accession, contig_name))
+                genes_and_functions_list.append(list_of_gene_attributes)
+            else:
+                # adding in "unknown annotation" if there is none
+                accession = "unknown-function"
+                contig_name = genes_in_contigs[counter]['contig']
+                list_of_gene_attributes.extend((counter,accession,contig_name))
+                genes_and_functions_list.append(list_of_gene_attributes)
+            counter = counter + 1
+
+        return genes_and_functions_list
+
+
+    def count_synteny(self, function_list, n):
+        """This method will count NGrams in contigs
+
+        This method will interate through a dict of contigs {contig_name: genes_and_functions_list} count NGrams
+        in each contig using a sliding window of size N. The final output will be a dictionary {ngram:count}
+
+        Parameters
+        ==========
+        function_list : list
+            A list of gene functions as they appear in the contig
+
+        n : int
+            A window size to extract a ngram
+
+        Returns
+        =======
+        NGramFreq_dict : dict
+            A dict where each key is list of genes that represent an ngram and the value is the count in that contig
+
+        Notes
+        =====
+        Future goal: Need to return counts for 1 contig at a time and
+        give back a dictionary with contig {contig_name: {ngram:count}}
+        """
+
+        NGramFreq_dict = {}
+        for i in range(0, len(function_list) - n + 1):
+            # window = sorted(function_list[i:i + n])
+            # extract window
+            window = function_list[i:i + n]
+
+            # order the window based arbitrarily based on the first gene in the synteny being smallest
+            original_order = window
+            flipped_order = window[::-1]
+            if original_order[0] < flipped_order[0]:
+                window = original_order
+            else:
+                window = flipped_order
+
+            ngram = tuple(window)
+
+            if not self.in_in_unknowns_mode and "unknown-function" in ngram: # conditional to record NGrams with unk functions
+                continue
+            else:
+                # if ngram is not in dictionary add it, if it is add + 1
+                if ngram in NGramFreq_dict:
+                    NGramFreq_dict[ngram] +=  1
+                else:
+                    NGramFreq_dict[ngram] = 1
+
+        return NGramFreq_dict
```

Comment:
very mixed :)

It should either be `NGramFreqDict` (which is a better class name than variable name) or `ngram_frequencies_dict` (which is what you mean here).